lib/cisco_node_utils/ace.rb

```diff
@@ -0,0 +1,225 @@
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # Ace - node utility class for Ace Configuration
+  class Ace < NodeUtil
+    attr_reader :acl_name, :seqno, :afi
+    # acl_name is name of acl
+    # seqno is sequence number of ace
+    # afi is either v4 or v6
+    def initialize(acl_name, seqno, afi)
+      fail TypeError unless
+        acl_name.is_a?(String) && seqno.is_a?(Integer) && afi.is_a?(String)
+      fail ArgumentError 'we expect ip or ipv6' unless
+        afi == 'ip' || afi == 'ipv6'
+      @acl_name = acl_name
+      @afi = afi
+      @seqno = seqno
+      @set_args = @get_args = { acl_name: @acl_name, seqno: @seqno, afi: @afi,
+                                src_port: '', dst_port: '',
+                                option_format: '' }
```

Comment:
Hi Chris
my acl,yaml file it has 
<src_port> and <dst_port> tag , i have to fill these in. 
if i leave it empty, does infra automatically fill it for me?